### PR TITLE
move methods from directives to functions

### PR DESCRIPTION
**Pull request by drewwells**
This moves the `is_digit` and `is_space` directives to methods.

### CODE REVIEW

**Comment by drewwells** [comment]
sourceforge!

**Comment by xzyfer** [comment]
Thanks @drewwells

File: src/json.cpp [hunk 1]
```diff
@@ -357,9 +357,8 @@ static void to_surrogate_pair(uint32_t unicode, uint16_t *uc, uint16_t *lc)
   *lc = (n & 0x3FF) | 0xDC00;
 }
 
-#define is_space(c) ((c) == '\t' || (c) == '\n' || (c) == '\r' || (c) == ' ')
-#define is_digit(c) ((c) >= '0' && (c) <= '9')
-
+bool is_space        (const char *c);
+bool is_digit        (const char *c);
```
Comment (mgreter) [inline src/json.cpp]: This declaration doesn't seem to match the implementation?

Comment (drewwells) [inline src/json.cpp]: In what way? I declared it as static bool and some of the windows and os x
builds failed. This declaration passed all builds though.

Comment (xzyfer) [inline src/json.cpp]: @drewwells 

```
bool is_digit (const char *c)
```
vs
```
bool is_space(const char& c)
```

My mistake for noticing this.

Comment (drewwells) [inline src/json.cpp]: Ahh good catch, I'll PR a fix.

Comment (xzyfer) [inline src/json.cpp]: I'm working on it in https://github.com/sass/libsass/pull/1395.

There are other inconsistencies I want to fix.

**Pull request merged**